bug: oomcli snapshot failed

recurrent:

```
(base) ➜  oomstore git:(fix-stream-join) ✗ oomcli init
(base) ➜  oomstore git:(fix-stream-join) ✗ oomcli apply -f /tmp/oomstore/meta.yaml
applied
(base) ➜  oomstore git:(fix-stream-join) ✗ oomcli get meta entity -o yaml
items:
    - kind: Entity
      name: device
      description: device
      groups:
        - name: phone
          category: batch
          description: phone
          features:
            - name: price
              value-type: int64
              description: price
            - name: model
              value-type: string
              description: model
    - kind: Entity
      name: user
      description: user
      groups:
        - name: student
          category: batch
          description: student
          features:
            - name: age
              value-type: int64
              description: age
        - name: user-click
          category: stream
          description: user click post feature
          features:
            - name: last_5_click_posts
              value-type: string
              description: user last 5 click posts
            - name: number_of_user_started_posts
              value-type: int64
              description: number of posts that users stared today
(base) ➜  oomstore git:(fix-stream-join) ✗ oomcli push --entity-key 1 --group user-click --features last_5_click_posts=1,2,3,4,5 --features number_of_user_started_posts=10
succeeded
(base) ➜  oomstore git:(fix-stream-join) ✗ oomcli snapshot user-click
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/oom-ai/oomstore/pkg/oomstore.(*OomStore).Snapshot(0xc0003c30c0, {0x29618e8, 0xc0000440d0}, {0x7ffeefbff368, 0x0})
	/Users/lianxm/github.com/oomstore/pkg/oomstore/snapshot.go:40 +0x3e8
github.com/oom-ai/oomstore/oomcli/cmd.glob..func34(0x36c2080, {0xc000a548b0, 0x1, 0x1})
	/Users/lianxm/github.com/oomstore/oomcli/cmd/snapshot.go:24 +0xe5
github.com/spf13/cobra.(*Command).execute(0x36c2080, {0xc000a54880, 0x1, 0x1})
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x36c1b80)
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/oom-ai/oomstore/oomcli/cmd.Execute()
	/Users/lianxm/github.com/oomstore/oomcli/cmd/root.go:32 +0x25
main.main()
	/Users/lianxm/github.com/oomstore/oomcli/main.go:6 +0x17
(base) ➜  oomstore git:(fix-stream-join) ✗
```

--- a/pkg/oomstore/snapshot.go
+++ b/pkg/oomstore/snapshot.go
@@ -29,6 +29,12 @@ func (s *OomStore) Snapshot(ctx context.Context, groupName string) error {
 			return err
 		}
 	}
+
+	// re-fetch revision
+	revisions, err = s.metadata.ListRevision(ctx, &group.ID)
+	if err != nil {
+		return err
+	}
 	for i, revision := range revisions {
 		if revision.SnapshotTable != "" {
 			continue

